From kristen.c.accardi@intel.com Thu Aug  9 16:09:43 2007
From: Kristen Carlson Accardi <kristen.c.accardi@intel.com>
From: kristen.c.accardi@intel.com
Date: Thu, 09 Aug 2007 16:09:32 -0700
Subject: PCI Hotplug: pciehp: Request control over PCI Express Capability as well as Native hotplug
To: gregkh@suse.de
Cc: Kristen Carlson Accardi <kristen.c.accardi@intel.com>
Message-ID: <20070809230943.119990208@intel.com>

From: Kristen Carlson Accardi <kristen.c.accardi@intel.com>

According to the PCI firmware spec (3.0), the OS must claim control
over the PCI Express Capability bits in addition to the PCI Express
Native Hot Plug feature when executing _OSC.

Signed-off-by: Kristen Carlson Accardi <kristen.c.accardi@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/hotplug/pciehp_hpc.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/pci/hotplug/pciehp_hpc.c
+++ b/drivers/pci/hotplug/pciehp_hpc.c
@@ -1138,6 +1138,7 @@ int pciehp_acpi_get_hp_hw_control_from_f
 		dbg("Trying to get hotplug control for %s \n",
 			(char *)string.pointer);
 		status = pci_osc_control_set(handle,
+				OSC_PCI_EXPRESS_CAP_STRUCTURE_CONTROL |
 				OSC_PCI_EXPRESS_NATIVE_HP_CONTROL);
 		if (status == AE_NOT_FOUND)
 			status = acpi_run_oshp(handle);
